commit bf868ac056512b3ccd9a1a6eebfc62cd762c6bb9
Author: Andrew Bagshaw <abagshaw@hotmail.ca>
Date:   Tue Apr 11 17:45:35 2017 -0700

    Check to see if summary flag exists.
    
    TensorBoard is awesome...but it's not for everyone :) - don't use
    TensorBoard features if "summary" flag doesn't exist (mostly a fix for
    return_predict usage of darkflow)

diff --git a/net/build.py b/net/build.py
index 464ee54..2abfd7a 100644
--- a/net/build.py
+++ b/net/build.py
@@ -104,9 +104,10 @@ class TFNet(object):
 			cfg['device_count'] = {'GPU': 0}
 
 		if self.FLAGS.train: self.build_train_op()
-
-		self.summary_op = tf.summary.merge_all()
-		self.writer = tf.summary.FileWriter(self.FLAGS.summary + 'train')
+		
+		if self.FLAGS.summary is not None:
+			self.summary_op = tf.summary.merge_all()
+			self.writer = tf.summary.FileWriter(self.FLAGS.summary + 'train')
 		
 		self.sess = tf.Session(config = tf.ConfigProto(**cfg))
 		self.sess.run(tf.global_variables_initializer())
@@ -115,8 +116,9 @@ class TFNet(object):
 		self.saver = tf.train.Saver(tf.global_variables(), 
 			max_to_keep = self.FLAGS.keep)
 		if self.FLAGS.load != 0: self.load_from_ckpt()
-
-		self.writer.add_graph(self.sess.graph)
+		
+		if self.FLAGS.summary is not None:
+			self.writer.add_graph(self.sess.graph)
 
 	def savepb(self):
 		"""
